Update installation and ROCPD documentation

dgaliffiAMD opened the pull request:
## Motivation

Update documentation for rocm 7.1 and rocm-systems

## Technical Details

All references to rocprofiler-systems as a standalone repo were updated to rocm-systems and to the new directory structure.
Installation and Quick-Start pages were merged.
Added section to "understanding-rocprof-sys-output.rst" for rocpd output

dgaliffiAMD committed:
Updating install doc page

dgaliffiAMD committed:
Removing the Quick Start page

dgaliffiAMD committed:
Add documentation for rocpd output

dgaliffiAMD committed:
Update links to reference rocm-systems repo

dgaliffiAMD committed:
Update README.md

Installation instructions references ROCm Docs link.

dgaliffiAMD committed:
Updated git clone instructions

Back to using https to clone the repository

dgaliffiAMD committed:
Fix formatting

dgaliffiAMD committed:
Update projects/rocprofiler-systems/docs/how-to/understanding-rocprof-sys-output.rst

dgaliffiAMD committed:
Add reference to "rocpd" section to the "Profiling Python" section

dgaliffiAMD committed:
Update CONTRIBUTING.md

dgaliffiAMD committed:
For ROCPD, document minimum version of SDK.

dgaliffiAMD committed:
Update CHANGELOGS

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

prbasyal-amd commented on projects/rocprofiler-systems/CHANGELOG.md:
```diff
@@ -10,10 +10,21 @@ Full documentation for ROCm Systems Profiler is available at [https://rocm.docs.
 
 - ``ROCPROFSYS_ROCM_GROUP_BY_QUEUE`` configuration setting to allow grouping of events by hardware queue, instead of the default grouping.
 - Support for rocPD database output with the `ROCPROFSYS_USE_ROCPD` configuration setting.
```
```suggestion
- Support for `rocpd` database output with the `ROCPROFSYS_USE_ROCPD` configuration setting.
```

prbasyal-amd commented on projects/rocprofiler-systems/CHANGELOG.md:
```diff
@@ -10,10 +10,21 @@ Full documentation for ROCm Systems Profiler is available at [https://rocm.docs.
 
 - ``ROCPROFSYS_ROCM_GROUP_BY_QUEUE`` configuration setting to allow grouping of events by hardware queue, instead of the default grouping.
 - Support for rocPD database output with the `ROCPROFSYS_USE_ROCPD` configuration setting.
+- Support for profiling Pytorch workloads using the rocPD output database.
```
```suggestion
- Support for profiling PyTorch workloads using the `rocpd` output database.
```

dgaliffiAMD committed:
Merge branch 'develop' into users/dgaliffi/doc-updates

dgaliffiAMD committed:
Update CHANGELOG.md

Updated based on feedback from docs team

prbasyal-amd commented on projects/rocprofiler-systems/CHANGELOG.md:
```diff
@@ -10,10 +10,21 @@ Full documentation for ROCm Systems Profiler is available at [https://rocm.docs.
 ## Changed
 
 - Updated the grouping of "kernel dispatch" and "memory copy" events in Perfetto traces. They are now grouped together by HIP Stream rather than separately and by hardware queue.
+- Updated PAPI module to v7.2.0b2
```
```suggestion
- Updated PAPI module to v7.2.0b2.
```

prbasyal-amd commented on projects/rocprofiler-systems/CHANGELOG.md:
```diff
@@ -10,10 +10,21 @@ Full documentation for ROCm Systems Profiler is available at [https://rocm.docs.
 
 - Updated the grouping of "kernel dispatch" and "memory copy" events in Perfetto traces. They are now grouped together by HIP Stream rather than separately and by hardware queue.
+- Updated PAPI module to v7.2.0b2
+- Using ROCprofiler-SDK for tracing OMPT API calls.
```
```suggestion
- ROCprofiler-SDK is now used for tracing OMPT API calls.
```

prbasyal-amd commented on projects/rocprofiler-systems/CONTRIBUTING.md:
```diff
@@ -58,13 +58,13 @@ By creating a pull request, you agree to the statements made in the [code licens
-* Set the original repository URL as the remote upstream using `git remote add upstream https://github.com/ROCm/rocprofiler-systems` (or `git remote set-url upstream https://github.com/ROCm/rocprofiler-systems`)
+* It is recommended to [fork](https://github.com/ROCm/rocm-systems/fork) the repository.
+* Clone your forked repository: `git clone https://github.com/ROCm/<yourgithub-id>/rocm-systems.git`
+* Navigate to the project directory: `cd rocm-systems`
```
```suggestion
* Navigate to the project directory: `cd rocm-systems`
```
Should this be updated? The path for ROCm Systems Profiler specific Contribution.md is: rocm-systems/projects/rocprofiler-systems

prbasyal-amd commented on projects/rocprofiler-systems/docs/how-to/understanding-rocprof-sys-output.rst:
```diff
@@ -358,7 +319,56 @@ set ``ROCPROFSYS_OUTPUT_PREFIX="%argt%-"``, and let ROCm Systems Profiler cleanl
+
+Use the ``ROCPROFSYS_USE_ROCPD`` setting to trigger the ROCm Systems Profiler to output a
+SQLite3 database. The ROCm Profiling Data (or ``rocpd``) database will soon be the default output
+format. This output format requires ROCprofiler-SDK v1.0 or higher, which is included in ROCm v7.0.
```
```suggestion
format. The rocpd output format requires ROCprofiler-SDK 1.0.0 or higher.
```

dgaliffiAMD replied:
Applying the suggestions, but don't you think it useful to also include the version of ROCm that ROCProfiler-SDK was included in?

prbasyal-amd commented on projects/rocprofiler-systems/docs/how-to/understanding-rocprof-sys-output.rst:
```diff
@@ -358,7 +319,56 @@ set ``ROCPROFSYS_OUTPUT_PREFIX="%argt%-"``, and let ROCm Systems Profiler cleanl
+SQLite3 database. The ROCm Profiling Data (or ``rocpd``) database will soon be the default output
+format. This output format requires ROCprofiler-SDK v1.0 or higher, which is included in ROCm v7.0.
+
+Features
```
```suggestion
Features of rocpd format
```

prbasyal-amd commented on projects/rocprofiler-systems/docs/how-to/understanding-rocprof-sys-output.rst:
```diff
@@ -358,7 +319,56 @@ set ``ROCPROFSYS_OUTPUT_PREFIX="%argt%-"``, and let ROCm Systems Profiler cleanl
+format. This output format requires ROCprofiler-SDK v1.0 or higher, which is included in ROCm v7.0.
+
+Features
+++++++++
```
```suggestion
-----------------------------------------------
```

dgaliffiAMD committed:
Update CONTRIBUTING.md

dgaliffiAMD committed:
Update CONTRIBUTING.md.

Simplify and remove setup information overlapping with the "rocm-systems" contributing documentation.

prbasyal-amd commented on projects/rocprofiler-systems/CONTRIBUTING.md:
```diff
@@ -6,24 +6,13 @@
 
 # Contributing to rocprofiler-systems #
```
```suggestion
# Contributing to ROCm Systems Profiler #
```

prbasyal-amd commented on projects/rocprofiler-systems/CONTRIBUTING.md:
```diff
@@ -6,24 +6,13 @@
-4. [Coding Style](#coding-style)
-5. [Code License](#code-license)
-6. [References](#references)
+We welcome contributions to rocprofiler-systems.  Contributions at a basic level must conform to the MIT license and pass code test requirements (i.e. ctest). The author must also be able to respond to comments/questions on the PR and make any changes requested.
```
```suggestion
Contributions to ROCm Systems Profiler are encouraged and welcomed.  Contributions at a basic level must conform to the MIT license and pass the code test requirements (i.e., ctest). The author must also be able to respond to comments/questions on the PR and make any changes requested.
```

prbasyal-amd commented on projects/rocprofiler-systems/CONTRIBUTING.md:
```diff
@@ -33,55 +22,21 @@ Please use the GitHub Issues tab to notify us of issues.
-### Exceptions ###
-
-* If you believe your contribution does not fit the guidelines but is still valuable, please discuss it with the maintainers before submitting.
+Github issues are recommended for any significant change to the code base that adds a feature or fixes a non-trivial issue. If the code change is large without the presence of an issue (or prior discussion with AMD), the change may not be reviewed. Small fixes that fix broken behavior or other bugs are always welcome with or without an associated issue.
```
```suggestion
GitHub Issues are recommended for any significant change to the code base that adds a feature or fixes a non-trivial issue. If the code change is large without the presence of an issue (or prior discussion with AMD), the change may not be reviewed. Small fixes that fix broken behavior or other bugs are always welcome, with or without an associated issue.
```

prbasyal-amd reviewed (commented):
Minor comments

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

prbasyal-amd committed:
Update CHANGELOG.md

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>

prbasyal-amd commented on projects/rocprofiler-systems/docs/how-to/understanding-rocprof-sys-output.rst:
```diff
@@ -358,7 +319,56 @@ set ``ROCPROFSYS_OUTPUT_PREFIX="%argt%-"``, and let ROCm Systems Profiler cleanl
+
+Use the ``ROCPROFSYS_USE_ROCPD`` setting to trigger the ROCm Systems Profiler to output a
+SQLite3 database. The ROCm Profiling Data (or ``rocpd``) database will soon be the default output
+format. The rocpd output format requires ROCprofiler-SDK 1.0.0 or higher.
```
```suggestion
format. To output in `rocpd` format, ROCProfiler-SDK version 1.0.0 or later is required (introduced in ROCm 7.0.0).
```

prbasyal-amd reviewed (approved):
Minor comment

dgaliffiAMD committed:
Apply suggestion from @prbasyal-amd

Co-authored-by: Pratik Basyal <pratik.basyal@amd.com>
